packages/console/src/pages/ApplicationDetails/components/AdvancedSettings.tsx

```diff
@@ -121,54 +121,65 @@ function AdvancedSettings({ applicationType, oidcConfig }: Props) {
           />
         </FormField>
       )}
-      {applicationType !== ApplicationType.MachineToMachine && (
-        <>
-          <FormField title="application_details.rotate_refresh_token">
-            <Switch
-              label={
-                <Trans
-                  components={{
-                    a: (
-                      <TextLink
-                        href="https://docs.logto.io/docs/references/applications/#rotate-refresh-token"
-                        target="_blank"
-                      />
-                    ),
-                  }}
-                >
-                  {t('application_details.rotate_refresh_token_label')}
-                </Trans>
-              }
-              {...register('customClientMetadata.rotateRefreshToken')}
-            />
-          </FormField>
-          <FormField
-            title="application_details.refresh_token_ttl"
-            tip={t('application_details.refresh_token_ttl_tip')}
-          >
-            <TextInput
-              {...register('customClientMetadata.refreshTokenTtlInDays', {
-                min: {
-                  value: minTtl,
-                  message: ttlErrorMessage,
-                },
-                max: {
-                  value: maxTtl,
-                  message: ttlErrorMessage,
-                },
-                valueAsNumber: true,
-                validate: (value) =>
-                  value === undefined ||
-                  Number.isInteger(value) ||
-                  t('errors.should_be_an_integer'),
-              })}
-              placeholder="14"
-              // Confirm if we need a customized message here
-              error={errors.customClientMetadata?.refreshTokenTtlInDays?.message}
-            />
-          </FormField>
-        </>
-      )}
+      {/**
+        * Public clients (authentication method is none) are not allowed to disable refresh token
+        * rotation, so we don't show the option here.
```

Comment:
I think even for non-sender constraint public clients, we can still disable refresh token rotation. It is only by default that this option is enabled but it doesn't mean we can't turn it off.

Reply:
if we follow the default logic of oidc-provider, they should be able to turn off this

Reply:
Per offline discussion, the users should still be able to to turn it off even if it's SPA, although it is not recommended.